.github/workflows/test.yml

```
@@ -13,23 +13,31 @@ jobs:
     strategy:
       matrix:
         python: ["3.9", "3.10", "3.11"]
-        os: ["ubuntu-latest", "windows-latest"]
+        os: ["ubuntu-latest", "macos-latest", "windows-latest"]
         source: ["conda", "pip"]
     steps:
       - name: checkout
         uses: actions/checkout@v3
         with: { submodules: recursive }
       - name: setup conda environment
-        uses: conda-incubator/setup-miniconda@v2
+        uses: conda-incubator/setup-miniconda@v3
         with:
           python-version: "${{ matrix.python }}"
           miniforge-variant: Mambaforge
-      - name: install dependencies
+          miniforge-version: latest
+          environment-file: environment.yml
+          auto-update-conda: true
+      - name: list packages
         shell: bash -l {0}
         run: |
-          mamba env update --quiet -n test -f environment.yml
-          conda list
+          mamba info
+          mamba list
         if: ${{ matrix.source == 'conda' }}
+      - name: workaround for libtiff.5.dylib on macos
+        shell: bash -l {0}
+        run: |
+          ln -s /Users/runner/miniconda3/envs/test/lib/python${{ matrix.python }}/site-packages/PIL/../../../libtiff.6.dylib /Users/runner/miniconda3/envs/test/lib/python${{ matrix.python }}/site-packages/PIL/../../../libtiff.5.dylib
```

```suggestion
          ln -s /Users/runner/miniconda3/envs/test/lib/libtiff.6.dylib /Users/runner/miniconda3/envs/test/lib/libtiff.5.dylib
```

This feels quite wrong. This should be fixed upstream by fixing the pillow package in conda-forge.